base/src/main/java/com/mindera/skeletoid/threads/threadpools/NamedThreadFactory.kt

```diff
@@ -0,0 +1,72 @@
+package com.mindera.skeletoid.threads.threadpools
+
+import com.mindera.skeletoid.logs.LOG
+
+import java.util.Queue
+import java.util.concurrent.ArrayBlockingQueue
+import java.util.concurrent.ThreadFactory
+import java.util.concurrent.atomic.AtomicInteger
+
+/**
+ * Factory for threads that provides custom naming
+ */
+
+/**
+ * Default constructor
+ *
+ * @param threadNamePrefix The name of the ThreadPool
+ * @param maxThreads Max number of threads
+ */
+class NamedThreadFactory(private val threadNamePrefix: String, private val maxThreads: Int) :
+    ThreadFactory {
+
+    companion object {
+        private val LOG_TAG = "NamedThreadFactory"

```

Comment:
maybe `private const val` is more suitable since it is a literal and that type of notation is being used for the name?